src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerProjectSystem.cs

```diff
@@ -26,6 +26,12 @@
 
 namespace Microsoft.CodeAnalysis.LanguageServer.HostWorkspace;
 
+// base type LanguageServerProjectSystem (lacks mef attributes)
+// - abstract _workspaceFactory
+
+// derived: HostLanguageServerProjectSystem <--
+// derived: FileBasedProgramLanguageServerProjectSystem <--
```

Comment:
what is this?

Reply:
raw notes for how I plan on implementing the feature.